### PR TITLE
feat(cli): add self-contained executable build with PyInstaller

### PR DESCRIPTION
**Pull request by mattwise-42**
## Summary

Adds support for building self-contained OpenToken CLI executables using PyInstaller. The executables bundle Python, all dependencies, and the CLI code into standalone binaries for Linux, macOS (universal), and Windows — no Python installation required. Also adds `workflow_dispatch` triggers with version override support to all three publish workflows.

### Self-Contained Executable Build
- **build-opentoken-cli.yml**: GitHub Actions workflow for building CLI executables on all platforms
  - Triggers on every push to `main` and via manual `workflow_dispatch`
  - Runs on all three platforms (Linux, macOS, Windows) in parallel
  - Creates universal binaries for macOS (Intel + Apple Silicon) via `--target-arch universal2`
  - Includes smoke tests using the `tokenize` subcommand to verify executable functionality
  - Produces versioned zip files with platform identifiers
  - Uploads artifacts for each platform
- **opentoken-cli.spec**: PyInstaller spec file with proper dependency collection
  - Anchors paths to the spec file location (`SPECPATH`) for reproducible builds regardless of invocation directory
  - Collects pyarrow, pandas, csv2parquet, cryptography with all resources
  - Includes hidden imports for token definitions
  - Configures single-file executable with UPX compression
  - Entry script resolved via absolute path joined from `base_dir`
- **pyinstaller-requirements.txt**: Pins PyInstaller to 6.11.1

### Package Outputs
Versioned zip files containing standalone executables:
- `opentoken-cli-{version}-linux-x64.zip` — Linux 64-bit
- `opentoken-cli-{version}-macos-universal.zip` — macOS Universal Binary (Intel + Apple Silicon)
- `opentoken-cli-{version}-windows-x64.zip` — Windows 64-bit

Version is automatically extracted from `.bumpversion.cfg` (currently `2.0.0-alpha`).

### Token Registry Compatibility
- **token_registry.py**: Fixed module discovery to work in PyInstaller bundled environments
  - Multi-tier fallback chain:
    1. `pkgutil.iter_modules()` via `getattr(package, "__path__", None)` — normal Python filesystem
    2. `importlib.resources.files()` — packaged/bundled environments
    3. Hardcoded token list (`t1_token`–`t5_token`) — final fallback for edge cases
  - Removes filesystem path dependency; backward compatible with all existing environments
- **token_registry_test.py**: Comprehensive tests for all three fallback paths
  - Tests each fallback mechanism independently using `unittest.mock.patch`
  - Verifies consistency of token IDs and definition counts across all fallback methods

### Documentation Updates
- **README.md**: Added self-contained executable quickstart as the first installation option with separate bash and PowerShell code blocks
- **pages/quickstarts/cli-quickstart.md**: Added detailed download/install/run instructions for executables using `{version}` placeholders for zip and directory names; uses `package` subcommand in examples
- **pages/reference/cli.md**: Added Installation Options section listing all four install methods; fixed Python invocation to `python -m opentoken_cli.main`; separated Linux/macOS (bash) and Windows (PowerShell) command syntax blocks

### Infrastructure
- **.bumpversion.cfg**: Added file tracking entry for `pages/quickstarts/cli-quickstart.md` so version bumps automatically update the example zip/directory names in that file
- **.gitignore**: Added `smoke/` (CLI smoke test artifacts) and `.settings` (IDE config)
- **maven-publish.yml, python-publish.yml, docker-publish.yml**: Added `workflow_dispatch` triggers for manual emergency publishes; the `version` input is now properly wired to `release-context.yml` via the `release-tag` parameter (e.g., `format('v{0}', github.event.inputs.version)`) so manually provided versions are actually used

## Smoke Test Results ✅

All smoke tests **PASSED** on Linux ARM64 (Codespaces):

| Test | Status | Details |
|------|--------|---------|
| CSV to CSV tokenization | ✅ PASSED | 3 records → 15 tokens (5 rules × 3 records) |
| CSV to Parquet tokenization | ✅ PASSED | Valid Parquet output (10,298 bytes) |
| Token quality verification | ✅ PASSED | 0 blank tokens, 15/15 unique (100%) |
| Token registry verification | ✅ PASSED | All 5 rules (T1-T5) loaded correctly |

**Executable Details:**
- Size: 111 MB (self-contained, zero dependencies)
- Startup time: < 1 second
- Processing speed: 3 records in < 1 second

See `smoke/SMOKE_TEST_RESULTS.md` for detailed results.

## Testing

- [x] Verify PyInstaller builds complete successfully (Linux tested locally)
- [x] Test standalone executables work without Python installed
- [x] Verify smoke tests pass (all 4 test suites passing)
- [x] Confirm token registry works in bundled environment
- [x] Comprehensive unit tests for all three token registry fallback paths
- [x] CI Python tests passing (2562 tests across all Python 3.10/3.11/3.12 suites)
- [x] Java tests passing (405 tests)
- [ ] Test on macOS (Intel and Apple Silicon)
- [ ] Test on Windows
- [ ] Verify CI builds for all platforms via `build-opentoken-cli.yml`

## Files Changed

- `.bumpversion.cfg` (~4 lines) — add `cli-quickstart.md` version tracking
- `.github/workflows/build-opentoken-cli.yml` (117 lines, new) — PyInstaller build workflow
- `.github/workflows/docker-publish.yml` (~9 lines) — add `workflow_dispatch` with `release-tag` wiring
- `.github/workflows/maven-publish.yml` (~8 lines) — add `workflow_dispatch` with `release-tag` wiring
- `.github/workflows/python-publish.yml` (~8 lines) — add `workflow_dispatch` with `release-tag` wiring
- `.gitignore` (~3 lines) — add `smoke/` and `.settings`
- `README.md` (~14 lines) — add self-contained executable quickstart section
- `lib/python/opentoken-cli/opentoken-cli.spec` (51 lines, new) — PyInstaller spec file
- `lib/python/opentoken-cli/pyinstaller-requirements.txt` (1 line, new) — PyInstaller pin
- `lib/python/opentoken/src/main/opentoken/tokens/token_registry.py` (~37 lines) — multi-tier module discovery fallback
- `lib/python/opentoken/src/test/opentoken/tokens/token_registry_test.py` (~108 lines) — fallback tests
- `pages/quickstarts/cli-quickstart.md` (~61 lines) — executable install/run instructions
- `pages/reference/cli.md` (~36 lines) — installation options and corrected syntax examples

## Notes

**macOS Universal Binary**: Built using PyInstaller's `--target-arch universal2` flag. The binary will be approximately 2× larger than single-architecture builds but provides the best user experience across Intel and Apple Silicon Macs.

**Token Registry Changes**: Backward compatible — they improve module discovery for bundled environments while maintaining full compatibility with normal Python installations, Java sync expectations, and existing test suites.

**Publish Workflow Fix**: The `release-tag` input is now properly passed to `release-context.yml` when `workflow_dispatch` is used, ensuring the manually supplied version is actually used rather than read from `.bumpversion.cfg`.

**Smoke Tests**: The `smoke/` directory contains automated tests and is excluded from git. Run `smoke/smoke_test.sh` to verify executable functionality.

### CODE REVIEW

**Review by Sarankdl — APPROVED** [review]

**Review by Sarankdl — APPROVED** [review]

File: .github/workflows/build-opentoken-cli.yml [hunk 1]
```diff
@@ -0,0 +1,116 @@
+name: Build OpenToken CLI (PyInstaller)
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - main
+
```
[the remark this answers is not in the document]
Comment (mattwise-42) [reply .github/workflows/build-opentoken-cli.yml:8]: it's as intended

File: pages/reference/cli.md [hunk 2]
```diff
@@ -6,6 +6,26 @@ layout: default
...
+- **macOS**: `opentoken-cli-{version}-macos-universal.zip` (Intel + Apple Silicon)
+- **Windows**: `opentoken-cli-{version}-windows-x64.zip`
+
+No dependencies required. Extract and run.
+
+### Other Options
+
+- **Docker**: Use `run-opentoken.sh` (Linux/Mac) or `run-opentoken.ps1` (Windows)
+- **Java**: Build from source with Maven, run `java -jar opentoken-cli-*.jar`
+- **Python**: Install via pip, run `opentoken` or `python -m opentoken_cli`
```
[the remark this answers is not in the document]
Comment (mattwise-42) [reply pages/reference/cli.md]: fixed

File: pages/quickstarts/cli-quickstart.md [hunk 3]
```diff
@@ -12,10 +12,69 @@ Run the OpenToken CLI end-to-end to generate tokens from a sample dataset in min
...
+chmod +x opentoken
+
+# Run the CLI
+./opentoken \
+  -i /path/to/sample.csv \
+  -o /path/to/output.csv \
+  -t csv \
+  -h "HashingKey" \
+  -e "Secret-Encryption-Key-Goes-Here."
+```
```
[the remark this answers is not in the document]
Comment (mattwise-42) [reply pages/quickstarts/cli-quickstart.md]: fixed
